props/saber_fett263_buttons.h

```diff
@@ -19,40 +41,87 @@
 // turn hilt to rotate through colors, click PWR to select/exit
 // if using COLOR_CHANGE_DIRECT each button press advances one Color at a time
 // Next Preset - click AUX while OFF
+// MULTI_PHASE Next Preset - hold AUX and TWIST while ON (use define to enable)
 // Previous Preset - hold AUX and click PWR while OFF
+// MULTI_PHASE Previous Preset - hold PWR and TWIST while ON (use define to enable)
 // Enter SA22C Volume Menu - hold and release AUX while OFF
 // Volume Up (10% increment, 100% max) - click PWR while in Volume Menu while OFF
 // Volume Down (10% increment) - click AUX while in Volume Menu while OFF
 // Exit Volume Menu - hold and release AUX while in Volume Menu while OFF
 
-#ifndef PROPS_SABER_FETT263_BUTTONS_H
-#define PROPS_SABER_FETT263_BUTTONS_H
+#ifndef PROPS_SABER_FETT263_G_BUTTONS_H
+#define PROPS_SABER_FETT263_G_BUTTONS_H
+
+#ifndef MOTION_TIMEOUT
+#define MOTION_TIMEOUT 60 * 15 * 1000
+#endif
+
+#ifndef LOCKUP_THRESHOLD_DELAY
+#define LOCKUP_THRESHOLD_DELAY 200
+#endif 
 
 #include "prop_base.h"
 #include "../sound/hybrid_font.h"
 
 #undef PROP_TYPE
-#define PROP_TYPE SaberFett263Buttons
+#define PROP_TYPE SaberFett263GButtons
 
 // The Saber class implements the basic states and actions
 // for the saber.
-class SaberFett263Buttons : public PropBase {
+class SaberFett263GButtons : public PropBase {
 public:
-SaberFett263Buttons() : PropBase() {}
-  const char* name() override { return "SaberFett263Buttons"; }
+SaberFett263GButtons() : PropBase() {}
+  const char* name() override { return "SaberFett263GButtons"; }
 
   // EVENT_SWING
   bool swinging_ = false;
   void Loop() override {
     PropBase::Loop();
+    // Swing On gesture control this portion allows fine tuning of speed needed to ignite
+    if(!SaberBase::IsOn() ) {
+      if (millis() - saber_off_time_ < MOTION_TIMEOUT) {
+        SaberBase::RequestMotion();
+        if (!swinging_ && fusor.swing_speed() > 250) {
+	// Edit '250' value in line above to change swing on sensitivity, 250 ~ 400 work best in testing
+           swinging_ = true;
+           Event(BUTTON_NONE, EVENT_SWING);
+        }
+        if (swinging_ && fusor.swing_speed() < 100) {
+           swinging_ = false;
+        }
+      }
+    }
+	  
     if (!swinging_ && fusor.swing_speed() > 250) {
       swinging_ = true;
       Event(BUTTON_NONE, EVENT_SWING);
     }
+	  
+    if(auto_lockup_on_) {
+       if (!swinging_ && fusor.swing_speed() > 120 && fusor.swing_speed() < 250 && millis() > clash_impact_millis_ + LOCKUP_THRESHOLD_DELAY) {
+          if (SaberBase::Lockup()) {
+             SaberBase::DoEndLockup();
+             SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+             auto_lockup_on_ = false;        
+          } 
+        }
+    }
+	  
     if (swinging_ && fusor.swing_speed() < 100) {
+      last_swing_ = millis();
```

Comment:
might be better to do:

if (swinging_) last_swing_ = millis();  (outside of this if statement)

Reply:
It's meant to be the time at the end of the swing or at clash so when swing ends this is the time set so if there are loud swings like accents or spins it doesn't activate the lockup/clash effect, only if the blade physically stops swinging can they trigger

Reply:
then why not just use !swinging_ ?